src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -92,6 +92,129 @@ def _get_symbol_table(node):
 
 
+def check_bound(array, dim, index, operator):
+    '''A Fortran array section with a missing lower bound implies the
+    access starts at the first element and a missing upper bound
+    implies the access ends at the last element e.g. a(:,:)
+    accesses all elements of array a and is equivalent to
+    a(lbound(a,1):ubound(a,1),lbound(a,2):ubound(a,2)). The PSyIR
+    does not support the shorthand notation, therefore the lbound
+    and ubound operators are used in the PSyIR.
+
+    This utility function checks that shorthand lower or upper
+    bound Fortran code is captured as longhand lbound and/or
+    ubound functions as expected in the PSyIR.
+
+    The supplied "array" argument is assumed to be an Array node,
+    the specified dimension "dim" is assumed to be a Range node
+    and the supplied range index "index" is assumed to be valid.
+
+    :param array: the node to check.
+    :type array: :py:class:`pysclone.psyir.node.array`
+    :param int dim: the dimension index to check.
+    :param int index: the index of the range to check (0 is the \
+        lower bound, 1 is the upper bound).
+    :param operator: the expected operator.
+    :type operator: \
+        :py:class:`psyclone.psyir.nodes.binaryoperation.Operator.LBOUND` \
+        or :py:class:`psyclone.psyir.nodes.binaryoperation.Operator.UBOUND`
+
+    :raises InternalError: if the supplied array does not have the \
+    expected properties.
+
+    '''
+    try:
+        bound = array.children[dim-1].children[index]
+        assert isinstance(bound, BinaryOperation), \
+            "Expecting BinaryOperation but found '{0}'." \
+            "".format(type(bound).__name__)
+        assert bound.operator == operator, \
+            "Expecting operator to be '{0}', but found '{1}'." \
+            "".format(str(operator), str(bound.operator))
+        reference = bound.children[0]
+        literal = bound.children[1]
+        assert isinstance(reference, Reference), \
+            "Expecting Reference but found '{0}'." \
+            "".format(type(reference).__name__)
+        assert reference.symbol is array.symbol, \
+            "Expecting Reference symbol '{0}' to be the same as array " \
+            "symbol '{1}'.".format(reference.symbol.name, array.symbol.name)
+        assert isinstance(literal, Literal), \
+            "Expecting Literal but found '{0}'." \
+            "".format(type(literal).__name__)
+        assert literal.datatype == DataType.INTEGER, \
+            "Expecting integer datatype but found '{0}'." \
+            "".format(literal.datatype)
+        assert literal.value == str(dim), \
+            "Expecting literal value '{0}' to be the same as the current " \
+            "array dimension '{1}'.".format(literal.value, str(dim))
+    except AssertionError as excinfo:
+        raise InternalError("psyir/frontend/fparser2.py:check_bound():"
+                            + str(excinfo))
+
+
+def check_literal(node, dim, index, value):
```

Comment:
Again, this is an internal utility. Please name appropriately and, if possible, move within the `check_range` routine.

Reply:
I've added an underscore. Again this routine can't be moved into `check_range` as it used for tests in `fparser2_test.py`.